Plugins/Json/include/Acts/Plugins/Json/SurfaceJsonConverter.hpp

```diff
@@ -0,0 +1,67 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Plugins/Json/AlgebraJsonConverter.hpp"
+#include "Acts/Plugins/Json/SurfaceBoundsJsonConverter.hpp"
+#include "Acts/Surfaces/Surface.hpp"
+
+#include <string>
+#include <vector>
+
+#include <nlohmann/json.hpp>
+
+// Custom Json encoder/decoders. Naming is mandated by nlohman::json and thus
+// can not match our naming guidelines.
+namespace Acts {
+
+static std::vector<std::string> surfaceTypes = {
+    "ConeSurface",  "CylinderSurface", "DiscSurface",       "PerigeeSurface",
+    "PlaneSurface", "StrawSurface",    "CurvilinearSurface"};
+
+/// Non-contextual conversion of a surface
+///
+/// @note it will take the default context
+void to_json(nlohmann::json& j, const Surface& surface);
+
+/// Contextual conversion of a surface
+///
+/// @param j the json to be filled
+/// @param surface the surface to be converted
+/// @param gctx the geometry context for this
+void toJson(nlohmann::json& j, const Surface& surface,
+            const Acts::GeometryContext& gctx);
+
+/// Converstion to Surface from jsonn
+///
+/// @param j the read-in json object
+///
+/// @return a shared_ptr to a surface object for type polymorphism
+std::shared_ptr<Surface> surfaceFromJson(const nlohmann::json&);
```

Comment:
You could also instead of `surfaceFromJson` use `from_json(const nlohmann::json& j, std::shared_ptr<Surface>& t)` not sure it would be better but that is an option.